src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java

```diff
@@ -167,75 +168,26 @@ public static boolean sendEmailViaSendGrid(String toEmail, String subject, Strin
     }
 
     /**
-     * Ensures that the provided phone number is a domestic mobile number.
-     * Returns a 400-Bad request status if that is not the case.
+     * @return true if the specified phoneNumber matches the configured regexp;
+     *   false if the phoneNumber is null, empty string, or PHONE_NUMBER_REGEXP has not been configured.
      */
-    public static com.twilio.rest.lookups.v1.PhoneNumber ensureDomesticPhoneNumber(spark.Request req, String phoneNumberString) {
-        if (phoneNumberString.startsWith("+1555555")) {
-            // For US fake 555 numbers (used in for UI and backend testing),
-            // create and return a PhoneNumber object from a minimal JSON string.
-            String lastFourDigits = phoneNumberString.substring("+1555555".length());
-            return com.twilio.rest.lookups.v1.PhoneNumber.fromJson(
-                "{\n" +
-                    "  \"national_format\": \"(555) 555-" + lastFourDigits + "\",\n" +
-                    "  \"phone_number\": \"" + phoneNumberString + "\"\n" +
-                    "}", new ObjectMapper()
-            );
-        }
+    public static boolean isPhoneNumberIsValid(String phoneNumber) {
+        if (StringUtils.isBlank(phoneNumber)) return false;
 
-        com.twilio.rest.lookups.v1.PhoneNumber phoneNumber = null;
-        try {
-            Twilio.init(TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN);
-            phoneNumber = com.twilio.rest.lookups.v1.PhoneNumber.fetcher(
-                new PhoneNumber(phoneNumberString))
-                .setCountryCode(COUNTRY_CODE)
-                .setType(List.of("carrier"))
-                .fetch();
-        } catch (ApiException apiException) {
-            // Handle 404 response - corresponds to invalid number.
-            // In that case we return a 400 bad request response to the requester.
-            if (apiException.getStatusCode() == 404) {
-                logMessageAndHalt(
-                    req,
-                    HttpStatus.BAD_REQUEST_400,
-                    "Phone number format is invalid."
-                );
-            } else {
-                logMessageAndHalt(
-                    req,
-                    HttpStatus.INTERNAL_SERVER_ERROR_500,
-                    "Error validating phone number format."
-                );
-            }
-        } catch (Exception e) {
-            logMessageAndHalt(
-                req,
-                HttpStatus.INTERNAL_SERVER_ERROR_500,
-                "Error validating phone number format."
-            );
-        }
+        // If no regex string/pattern is configured, assume phone number is valid.
+        if (PHONE_PATTERN == null) return true;
```

Comment:
This will go away once we hardcode the regex.

Reply:
I don't think we want to hardcode the regex.